### PR TITLE
Feat/waves

### PR DESCRIPTION
**Pull request by aurelenc**
- Add wave system
- Auto next wave after a defined time
- Auto next wave after all enemies are dead
- The number of enemies increase up to wave 31, after that, the difficulty for the players is to survive without being able to regen their health

### CODE REVIEW

File: src/utils/Window.cpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #ifdef CLIENT_COMPILATION_MODE
-#include "Window.hpp"
+    #include "Window.hpp"
```
Comment (GuyomT) [inline src/utils/Window.cpp:2]: What is this tab ? PR refused.

**Review by GuyomT — APPROVED** [review]
still searching for the doc
LGTM

File: src/ecs/SparseArray.hpp [hunk 2]
```diff
@@ -106,7 +106,11 @@ namespace ecs
             return _data[pos];
         }
 
-        void erase(sizeType pos) { _data[pos] = std::nullopt; }
+        void erase(sizeType pos)
+        {
+            if (pos < _data.size() && _data[pos])
```
Comment (Cavonstavant) [inline src/ecs/SparseArray.hpp:111]: is it safe to check the contents of `_data[pos]` on the same condition where you are checking out of bounds ?
Also, why are you checking its content ?

File: src/utils/Window.cpp (hunk 1 above)
Comment (aurelenc) [reply src/utils/Window.cpp:2]: its 4 spaces

**Review by Cavonstavant — APPROVED** [review]

**Pull request merged**